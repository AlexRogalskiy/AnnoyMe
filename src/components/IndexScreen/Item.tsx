import React from "react";
import { Text, View } from "react-native";
import { Annoy } from "../../types";
import styles from "./styles";

interface Props {
  item: Annoy;
}

export default ({ item }: Props) => (
  <View
    style={[
      styles.itemContainer,
      item.isActiveNow ? styles.itemActive : styles.itemInactive,
    ]}
  >
    <Text style={styles.itemTitle}>{item.title}</Text>
  </View>
);
