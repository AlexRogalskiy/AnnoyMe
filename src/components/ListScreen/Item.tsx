import React from "react";
import { Text, TouchableOpacity } from "react-native";
import TaskWithStatus from "../../dto/TaskWithStatus";
import styles from "./styles";

interface Props {
  task: TaskWithStatus;

  onPress: () => void;

  testID?: string;
}

/** Single task item for a list. */
export default ({ task, onPress }: Props) => (
  <TouchableOpacity
    style={[
      styles.itemContainer,
      task.isActive ? styles.itemActive : styles.itemInactive,
    ]}
    onPress={onPress}
    testID="task-item"
  >
    <Text style={styles.itemTitle}>{task.title}</Text>
  </TouchableOpacity>
);
