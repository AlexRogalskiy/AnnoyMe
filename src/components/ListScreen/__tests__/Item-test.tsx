import React from "react";
import renderer, { act } from "react-test-renderer";
import Item from "../Item";

it("Item renders active task", () => {
  const task = {
    id: "test",
    created: new Date(),
    title: "Test task!",
    schedule: {},
    isActive: true,
  };

  const tree = renderer
    .create(<Item task={task} onPress={() => undefined} />)
    .toJSON();

  expect(tree).toMatchSnapshot("Item-active");
});

it("Item renders inactive task", () => {
  const task = {
    id: "test",
    created: new Date(),
    title: "Test inactive task!",
    schedule: {},
    isActive: false,
  };

  const tree = renderer
    .create(<Item task={task} onPress={() => undefined} />)
    .toJSON();

  expect(tree).toMatchSnapshot("Item-inactive");
});

it("Item handless press", () => {
  const task = {
    id: "test",
    created: new Date(),
    title: "Test task!",
    schedule: {},
    isActive: true,
  };

  const onPress = jest.fn(() => undefined);

  const component = renderer.create(<Item task={task} onPress={onPress} />)
    .root;

  const touchable = component.findByProps({ testID: "task-item" });
  expect(touchable).toBeTruthy();

  act(() => touchable.props.onPress());
  expect(onPress.mock.calls).toEqual([[]]);
});
