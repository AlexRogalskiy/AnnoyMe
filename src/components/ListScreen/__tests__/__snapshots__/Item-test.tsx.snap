// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders active task: Item-active 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#9ccc65",
      "borderBottomColor": "#eeeeee",
      "borderBottomWidth": 2,
      "opacity": 1,
    }
  }
  testID="task-item"
>
  <Text
    style={
      Object {
        "fontWeight": "400",
        "padding": 20,
      }
    }
  >
    Test task!
  </Text>
</View>
`;

exports[`Item renders inactive task: Item-inactive 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#eeeeee",
      "borderBottomWidth": 2,
      "opacity": 1,
    }
  }
  testID="task-item"
>
  <Text
    style={
      Object {
        "fontWeight": "400",
        "padding": 20,
      }
    }
  >
    Test inactive task!
  </Text>
</View>
`;
